Plutarch/Rational.hs

```diff
@@ -11,12 +12,33 @@ module Plutarch.Rational (
 import Plutarch.Prelude
 
 import Data.Ratio (denominator, numerator)
-import Plutarch (PlutusType (..))
+import Plutarch (PlutusType (..), punsafeCoerce)
 import Plutarch.Bool (PEq (..), POrd (..), pif)
+import Plutarch.Builtin (
+  PAsData,
+  PBuiltinList,
+  PIsData (..),
+  pasInt,
+  pasList,
+  pforgetData,
+ )
 import Plutarch.Integer (PInteger, PIntegral (pdiv, pmod))
+import Plutarch.List (PListLike (pcons, phead, pnil, ptail), pmap)
 
 data PRational s = PRational (Term s PInteger) (Term s PInteger)
 
+instance PIsData PRational where
+  pfromData x = pListToRat #$ pmap # pasInt #$ pasList # pforgetData x
+  pdata x =
+    (punsafeCoerce :: Term _ (PAsData (PBuiltinList (PAsData PInteger))) -> Term _ (PAsData PRational)) $
+      pdata $ pmap # plam pdata #$ pRatToList # x
```

Comment:
Why not just do `pdata $` instead of `plam pdata #$`?

Reply:
Otherwise this looks fine actually :+1: , however, there is a design choice to be made here, between this approach and another approach. @TotallyNotChase

Reply:
#56

Reply:
I just noticed these aren't hoisted.